listenbrainz/webserver/views/playlist.py

```diff
@@ -19,9 +19,9 @@ def load_playlist(playlist_mbid: str):
         raise BadRequest("Provided playlist ID is invalid: %s" % playlist_mbid)
 
     playlist = db_playlist.get_by_mbid(playlist_mbid, True)
-    # TODO: Allow playlist collaborators to access private playlist
-    if playlist is None or not playlist.public and (not current_user.is_authenticated or playlist.creator_id != current_user.id):
-        raise NotFound("Cannot find playlist: %s" % playlist_mbid)
+    if playlist is None or not playlist.public and not current_user.is_authenticated \
```

Comment:
because we actually have an object here, let's take advantage of it in order to ensure we don't mess this up in multiples 

in `class Playlist`:

```py
def is_visible_by(user):
    if self.public:
        return True
    if user.is_authenticated:
        if is owner
        elif is collaborator
        else: return False
```

However, I fear that this may be more difficult than expected, because it seems that `current_user` in the website is an object, and `user` in the API, based on the api token is a dictionary. This should be normalised somehow too, not in this fix

Reply:
Perhaps we can use the user ID instead since we have a way to get that both from the web and API views?

Reply:
yes, using the user id is fine, but we'll have to work out where to put the anonymous user check. Off the top of my head I don't know what `validate_auth_header` returns in the playlist get api method if you don't authenticate